pandas/tests/indexes/common.py

```diff
@@ -537,9 +538,11 @@ def test_delete_base(self, indices):
         assert result.equals(expected)
         assert result.name == expected.name
 
-        with pytest.raises((IndexError, ValueError)):
+        length = len(indices)
+        msg = r"delete\(\) missing 1 required positional argument: 'loc'"
+        with pytest.raises(TypeError, match=msg):
             # either depending on numpy version
-            indices.delete(len(indices))
+            indices.delete()
```

Comment:
I *think* you forgot to put ```length``` inside the ```indices.delete()```, on L545.

And I think that this also means you need to change the error message.

Reply:
i think @MomIsBestFriend is correct here, the unused `length` is also the source of the linting failure